src/qibolab/instruments/qm/config/config.py

```diff
@@ -48,30 +57,35 @@ def add_octave(self, device: str, connectivity: str):
             self.add_controller(connectivity)
             self.octaves[device] = Octave(connectivity)
 
-    def configure_dc_line(self, channel: QmChannel, config: OpxOutputConfig):
+    def configure_dc_line(
+        self, id: ChannelId, channel: DcChannel, config: OpxOutputConfig
+    ):
         controller = self.controllers[channel.device]
         controller.analog_outputs[channel.port] = AnalogOutput.from_config(config)
-        self.elements[str(channel.logical_channel.name)] = DcElement.from_channel(
-            channel
-        )
+        self.elements[id] = DcElement.from_channel(channel)
 
     def configure_iq_line(
-        self, channel: QmChannel, config: IqConfig, lo_config: OscillatorConfig
+        self,
+        id: ChannelId,
```

Comment:
Personally I avoid using `id` as it is a Python keyword, but since it's not causing any issues it should be fine.

Reply:
> Personally I avoid using `id` as it is a Python keyword, but since it's not causing any issues it should be fine.

That's what I did myself for a long time. Then, I considered the difference between a keyword and a built-in: it is shadowing the `id()` function, which I don't need in this restricted scope.

Reply:
It'd be funnier if `id` was `Callable` but since it's not we can probably survive with it.